drivers/dma/dma_stm32.c

```diff
@@ -54,26 +54,15 @@ static void dma_stm32_clear_stream_irq(struct device *dev, uint32_t id)
 	stm32_dma_clear_stream_irq(dma, id);
 }
 
-static void dma_stm32_irq_handler(void *arg)
+static void dma_stm32_irq_handler(struct device *dev, uint32_t id)
 {
-	struct device *dev = arg;
 	const struct dma_stm32_config *config = dev->config;
 	DMA_TypeDef *dma = (DMA_TypeDef *)(config->base);
 	struct dma_stm32_stream *stream;
 	uint32_t callback_arg;
-	int id;
 
-	for (id = 0; id < data->max_streams; id++) {
-		if (func_ll_is_active_tc[id](dma)) {
-			break;
-		}
-		if (stm32_dma_is_irq_happened(dma, id)) {
-			break;
-		}
-	}
-
-	if (id == data->max_streams) {
-		LOG_ERR("Unknown interrupt happened.");
+	if (id >= config->max_streams) {
```

Comment:
Maybe an assertion would be better? Is there any chance for this to happen?

Reply:
It should not happen AFAIK, so maybe a ASSERT is OK.